pkgs/by-name/tl/tldr/package.nix

```diff
@@ -1,60 +1,67 @@
 {
   lib,
-  stdenv,
   fetchFromGitHub,
-  curl,
-  libzip,
-  pkg-config,
   installShellFiles,
+  python3Packages,
 }:
 
-stdenv.mkDerivation rec {
+python3Packages.buildPythonApplication rec {
   pname = "tldr";
-  version = "1.6.1";
+  version = "3.3.0";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "tldr-pages";
-    repo = "tldr-c-client";
-    rev = "v${version}";
-    sha256 = "sha256-1L9frURnzfq0XvPBs8D+hBikybAw8qkb0DyZZtkZleY=";
+    repo = "tldr-python-client";
+    tag = version;
+    hash = "sha256-lc0Jen8vW4BNg784td1AZa2GTYvXC1d83FnAe5RZqpY=";
   };
 
-  buildInputs = [
-    curl
-    libzip
+  build-system = with python3Packages; [
+    setuptools
+    wheel
   ];
-  nativeBuildInputs = [
-    pkg-config
-    installShellFiles
+
+  dependencies = with python3Packages; [
+    termcolor
+    colorama
+    shtab
   ];
 
-  makeFlags = [
-    "CC=${stdenv.cc.targetPrefix}cc"
-    "LD=${stdenv.cc.targetPrefix}cc"
-    "CFLAGS="
+  nativeBuildInputs = [ installShellFiles ];
+
+  nativeCheckInputs = with python3Packages; [
+    pytest
   ];
 
-  installFlags = [ "PREFIX=$(out)" ];
+  checkPhase = ''
+    runHook preCheck
+    pytest -k 'not test_error_message'
+    runHook postCheck
+  '';
+
+  doCheck = true;
 
   postInstall = ''
```

Comment:
This should be conditional, see #308283

Reply:
Added `lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform)`